esx_service/vmdk_ops.py

```diff
@@ -1075,22 +1084,109 @@ def load_vmci():
    else:
        lib = CDLL(os.path.join(LIB_LOC, "libvmci_srv.so"), use_errno=True)
 
+def getLock(lockname):
+    '''
+    Create or return a existing lock identified by lockname.
+    '''
+    global rsrcLocks
+    global managerLock
+
+    if not rsrcLocks:
```

Comment:
Since `rsrcLocks` and `managerLock` are globals, why not create the `WeakValueDictionary` and `threading.Lock()` when defining the globals, rather than doing it lazily in the function? Doing it this way appears that it may have a race condition that causes the `managerLock` not to serve it's purpose. Of course maybe python prevents this type of thread pre-emption, I have no idea. However, I believe it's better to be on the safe side. Plus it's less code :)  The race condition is below:

Thread 1 and thread 2 both check `if not managerLock` on line 1097 and see it to be false. 
Thread 1 proceeds to create the lock on line 1098 then acquire it on line 1101. 
Thread 1 gets pre-empted.
Thread 2 now overwrites the global value `managerLock` with a new lock on line 1098 and acquire this new lock on line 1101.

`rsrcLocks` is no longer protected by the mutex as both threads have access to it. 2 different resource locks can be created and returned for the same lockname.

I suggest initializing globals at the top level to prevent this from occurring.